### PR TITLE
ui tweaks for hyperlinks

### PR DESCRIPTION
**Pull request by romainfrancois**
### Intent

Follow up for #11273 

### Approach

inspired from @GCRev changes in https://github.com/rstudio/rstudio/issues/11273#issuecomment-1151664672

https://user-images.githubusercontent.com/2625526/173048572-d6fd42ca-a8e1-4044-9ffe-2c678d5556a6.mov

https://user-images.githubusercontent.com/2625526/173048733-71e51785-f2a4-4751-9998-49f0d49edee5.mov

### Automated Tests

> Indicate whether this change includes unit tests or integration tests, or link a work item or pull request to add those tests to another repo. If the change cannot or will not be covered by a test, indicate why.

### QA Notes

> Add additional information for QA on how to validate the change, paying special attention to the level of risk, adjacent areas that could be affected by the change, and any important contextual information not present in the linked issues. 

### Checklist

- [ ] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [ ] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [x] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [ ] This PR passes all local unit tests

### CODE REVIEW

**Comment by romainfrancois** [comment]
file hyperlinks no longer have a popup, we might reconsider if we have something interesting to show. 

vignette hyperlink get title *and* description (from the yaml front matter): 

image

File: src/cpp/session/modules/SessionHelp.R [hunk 1]
```diff
@@ -406,6 +406,31 @@ options(help_type = "html")
...
+{
+   description <- tryCatch(
+      {
+         v <- vignette(topic, package)
+
+         Dir <- v$Dir
+         File <- v$File
+         if (grepl("[.]Rmd$", File))
+         {
+            description <- rmarkdown::yaml_front_matter(file.path(Dir, "doc", File))$description
```
Comment (romainfrancois) [inline src/cpp/session/modules/SessionHelp.R:419]: Do we have something that would extract the yaml front matter without using `rmarkdown` ?

**Comment by GCRev** [comment]
This looks good to me! I'm glad we added theme-specific background styling for the pop-ups.

I'd still consider adding back in the "click to run" hint for code links so that it's totally clear what will happen when the user clicks on it.

**Comment by romainfrancois** [comment]
One thing I need to do is to handle the native event, so that e.g. we can close the popup with the keyboard etc ... but that can be a follow up pr.

**Comment by romainfrancois** [comment]
Merging this now, so that it may be in the build today. Will follow up.

**Pull request merged**